funding/manager.go

```diff
@@ -3527,6 +3655,45 @@ func (f *Manager) handleFundingLocked(peer lnpeer.Peer,
 		return
 	}
 
+	// If this is a taproot channel, then we'll need to map the received
+	// nonces to a nonce pair, and also fetch our pending nonces, which are
+	// required in order to make the channel whole.
+	var chanOpts []lnwallet.ChannelOpt
+	if channel.ChanType.IsTaproot() {
+		f.nonceMtx.Lock()
+		localNonce, ok := f.pendingMusigNonces[chanID]
+		if !ok {
+			// If there's no pending nonce for this channel ID,
+			// then we'll generate one now.
+			verNonce, err := lnwallet.NewMusigVerificationNonce(
```

Comment:
can use `channel.GenMusigNonces()` as is done for `createFundingLocked`

Reply:
So can't use it here as we don't yet have the full channel state machine created. So instead we need to call the function manually. Also spotted a bug here (will be covered by the eventual itests): it uses the current commitment height, but should actually use the _next_ commitment height. I think @ellemouton ran into this in the past.